app/code/core/Mage/SalesRule/Model/Resource/Coupon/Usage.php

```diff
@@ -61,7 +61,7 @@ public function updateCustomerCouponTimesUsed($customerId, $couponId, $decrement
 
         $timesUsed = $read->fetchOne($select, array(':coupon_id' => $couponId, ':customer_id' => $customerId));
 
-        if ($timesUsed !== false && $timesUsed > 0) {
```

Comment:
I guess there would be a problem if (somehow) the couponTimesUsed will be decremented several times until timeUsed less than 0. 
Maybe you should add $decrement to the if statement. Ensure to not descrement $timesUsed to less than 0

Reply:
The new code has a second if clause that would prevent this.